bottlerocket/agents/src/bin/ec2-resource-agent/ec2_provider.rs

```diff
@@ -235,8 +235,14 @@ where
     for instance_type in instance_types {
         info!("Using instance type '{}'", instance_type);
 
+        let mut flex_subnet_ids = spec.clone().configuration.subnet_ids;
```

Comment:
Should we do something like this?
```suggestion
        let mut flex_subnet_ids = spec.configuration.subnet_ids.clone();
```